cranelift/codegen/meta/src/shared/legalize.rs

```diff
@@ -212,8 +212,8 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
     // embedded as part of arguments), so use a custom legalization for now.
     narrow.custom_legalize(iconst, "narrow_iconst");
 
-    {
-        let inst = uextend.bind(I128).bind(I64);
+    for &(ty, ty_half) in &[(I128, I64), (I64, I32)] {
+        let inst = uextend.bind(ty).bind(ty_half);
```

Comment:
You can use `ty.half_width()`.

Reply:
That (and the other suggestion) doesn't seem to work for me--are these functions really available in `codegen/meta`?

Reply:
It doesn't seem they are available.